rfd/0057-automatic-user-provisioning.md

```diff
@@ -245,6 +245,35 @@ spec:
     logins: [ "{{external.username}}" ]
 ```
 
+## Update: allow the UID and GID of the created user to be specified
+
+This will require adding a new internal traits -- `host_user_uid` and
+`host_user_gid`, new allow/deny settings will be added to roles aswell
+for `host_user_uid` and `host_user_gid`.
```

Comment:
Yeah, I'm thinking it can be pretty confusing to make this a role setting. That's yet another field in the role and plus it conflict resolution gets tricky like you said.

I think that instead we maybe just take this from user's traits directly. So if a user has a trait, say, `teleport.dev/uid: 1234` then that's the UID used when provisioning a host user for them. It'd be somewhat similar to how it works with PAM now.

I'll discuss this UX on the product sync today.

Reply:
I've thought about this more today and got some questions about the exact customer's use-case which I posted in this ticket: https://github.com/gravitational/customer-sensitive-requests/issues/85#issuecomment-1631634641.

Let's wait for them to get back to us before finalizing the design, just want to make sure it'll work for their use-case.

Reply:
@lxea The customer [got back to us](https://github.com/gravitational/customer-sensitive-requests/issues/85#issuecomment-1635059658) and confirmed their users have unique usernames passed from IdP and that they have ability to set unique UID/GID as IdP attributes as well.

I think the approach with using traits I mentioned above should work fine then. Can you please update the RFD to that effect?

Reply:
I've updated this, when i was testing how SSO traits work, it created them as lists in teleport (you can see in my example here) is this expected/okay or is there some way to make it a key/singleton

Reply:
Should be fine. Let's just document that only the first entry will be significant.